### ISSUE TITLE
[RCS0034] Rule reports false positive on C# 12 Primary Constructors

### ISSUE THREAD

**Issue by MaksHladki** [dotnet/roslynator#1724]
### Product and Version Used
Roslynator Version: 4.14.1
IDE: Visual Studio 2022
Framework: .NET 8.0

### Description
The analyzer RCS0034 (Put type parameter constraint on its own line) reports a warning even when the constraint is placed on a new line, specifically when using C# 12 Primary Constructors.
It seems the analyzer does not account for the primary constructor parameter list appearing between the class declaration and the `where` clause.

### Steps to Reproduce

1. Create a class using C# 12 Primary Constructors with a generic type constraint.
```
public class MixedConstraintService<TKey, TValue>(TKey key, TValue value) where TKey : notnull where TValue : new()
{
    public TKey Key { get; } = key;
    public TValue Value { get; } = value;
}
```
2. Format the code so the where clause is on its own line (as required by RCS0034).
```
public class MixedConstraintService<TKey, TValue>(TKey key, TValue value)
    where TKey : notnull
    where TValue : new()
{
    public TKey Key { get; } = key;
    public TValue Value { get; } = value;
}
```

### Expected behavior
The analyzer should not report a warning because the `where` constraint is already on its own line.

### Actual behavior
The analyzer reports: _warning RCS0034: Put type parameter constraint on its own line_.
The CLI tool reports:  _Infinite loop detected: Reported diagnostics have been previously fixed_.

**Comment by josefpihrt** [comment]
Verified on current `main`: this was fixed by #1791. The exact repro from this issue passes with no RCS0034 diagnostic. Regression test added in the upcoming PR.
